pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -134,48 +138,49 @@ func pythonDetector(args []string, envs map[string]string) Instrumentation {
 	return None
 }
 
+// isNodeInstrumented parses the provided `os.File` trying to find an
+// entry for APM NodeJS instrumentation. Returns true if finding such
+// an entry, false otherwise.
+func isNodeInstrumented(f *os.File) bool {
+	const readLimit = 16 * 1024 // Read 16KiB max
```

Comment:
Seems quite low. See for example https://techcommunity.microsoft.com/t5/azure/quot-the-package-json-file-is-too-large-size-limit-is-384000/m-p/3912192?lightbox-message-images-3912192=502609iED452F97F6AE2627 where someone is using package.json of 1 MiB or more.  The `canSafelyParse` function in the service name code uses 1 MiB (although it reads everything to memory).

Reply:
Changed the limit to 1MiB.